File: 백준/Silver/11047. 동전 0/동전 0.py
N, K = map(int, input().split())
coin = []
for _ in range(N):
  coin.append(int(input()))

sum = 0

for i in range(N-1, -1, -1):
  if K == 0:
    break
  sum += K // coin[i]
  K = K % coin[i]

print(sum)

File: 백준/Silver/11047. 동전 0/동전 0.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let [N, K] = input[0].split(" ").map(Number);
let coin = input.slice(1, N + 1).map(Number);

let result = 0;
for (let i = N - 1; i >= 0; i--) {
  result += parseInt(K / coin[i]);
  K = K % coin[i];
}

console.log(result);
